source/extensions/filters/http/adaptive_concurrency/adaptive_concurrency_filter.cc

```diff
@@ -32,13 +32,23 @@ Http::FilterHeadersStatus AdaptiveConcurrencyFilter::decodeHeaders(Http::HeaderM
     return Http::FilterHeadersStatus::StopIteration;
   }
 
-  rq_start_time_ = config_->timeSource().monotonicTime();
+  // When the deferred_sample_task_ object is destroyed, the time difference between its destruction
+  // and the request start time is measured as the request latency. This value is sampled by the
+  // concurrency controller either when encoding is complete or during destruction of this filter
+  // object.
+  const auto rq_start_time = config_->timeSource().monotonicTime();
+  deferred_sample_task_ = std::make_unique<Cleanup>([this, rq_start_time]() {
```

Comment:
nit: I think you can just do `[this, rq_start_time = config_->timeSource().monotonicTime()]`